lib/llvm/execution_engine.rb

```diff
@@ -182,7 +186,9 @@ class MCJITCompiler < ExecutionEngine
     def initialize(mod, options = {})
       options = {
         :opt_level             => 2, # LLVMCodeGenLevelDefault
-        :code_model            => 0, # LLVMCodeModelDefault
+        # code_model causes segfault with LLVMCodeModelDefault (0) so using
+        # LLVMCodeModelJITDefault (1) instead
+        :code_model            => 1,
```

Comment:
This backtrace suggests that it could be something about the run_function that I wrote, so here is the backtrace from commit 20a1a4c8c3889ad4d110afb3c5224ee40c086e24.

````
-- Ruby level backtrace information ----------------------------------------
/var/lib/gems/2.3.0/gems/minitest-5.3.5/lib/minitest.rb:55:in `block in autorun'
/var/lib/gems/2.3.0/gems/minitest-5.3.5/lib/minitest.rb:126:in `run'
/var/lib/gems/2.3.0/gems/minitest-5.3.5/lib/minitest.rb:149:in `__run'
/var/lib/gems/2.3.0/gems/minitest-5.3.5/lib/minitest.rb:149:in `map'
/var/lib/gems/2.3.0/gems/minitest-5.3.5/lib/minitest.rb:149:in `block in __run'
/var/lib/gems/2.3.0/gems/minitest-5.3.5/lib/minitest.rb:285:in `run'
/var/lib/gems/2.3.0/gems/minitest-5.3.5/lib/minitest.rb:306:in `with_info_handler'
/var/lib/gems/2.3.0/gems/minitest-5.3.5/lib/minitest.rb:317:in `on_signal'
/var/lib/gems/2.3.0/gems/minitest-5.3.5/lib/minitest.rb:286:in `block in run'
/var/lib/gems/2.3.0/gems/minitest-5.3.5/lib/minitest.rb:286:in `each'
/var/lib/gems/2.3.0/gems/minitest-5.3.5/lib/minitest.rb:287:in `block (2 levels) in run'
/var/lib/gems/2.3.0/gems/minitest-5.3.5/lib/minitest.rb:293:in `run_one_method'
/var/lib/gems/2.3.0/gems/minitest-5.3.5/lib/minitest.rb:759:in `run_one_method'
/var/lib/gems/2.3.0/gems/minitest-5.3.5/lib/minitest/test.rb:101:in `run'
/var/lib/gems/2.3.0/gems/minitest-5.3.5/lib/minitest/test.rb:276:in `with_info_handler'
/var/lib/gems/2.3.0/gems/minitest-5.3.5/lib/minitest.rb:317:in `on_signal'
/var/lib/gems/2.3.0/gems/minitest-5.3.5/lib/minitest/test.rb:102:in `block in run'
/var/lib/gems/2.3.0/gems/minitest-5.3.5/lib/minitest/test.rb:256:in `time_it'
/var/lib/gems/2.3.0/gems/minitest-5.3.5/lib/minitest/test.rb:103:in `block (2 levels) in run'
/var/lib/gems/2.3.0/gems/minitest-5.3.5/lib/minitest/test.rb:204:in `capture_exceptions'
/var/lib/gems/2.3.0/gems/minitest-5.3.5/lib/minitest/test.rb:106:in `block (3 levels) in run'
[snip]/ruby-llvm/test/call_test.rb:79:in `test_external_string'
[snip]/ruby-llvm/test/test_helper.rb:50:in `run_function_on_module'
[snip]/ruby-llvm/lib/llvm/execution_engine.rb:64:in `run_function'
[snip]/ruby-llvm/lib/llvm/execution_engine.rb:64:in `new'
[snip]/ruby-llvm/lib/llvm/execution_engine.rb:64:in `initialize'
[snip]/ruby-llvm/lib/llvm/execution_engine.rb:76:in `block in run_function'
[snip]/ruby-llvm/lib/llvm/execution_engine.rb:76:in `run_function'

-- C level backtrace information -------------------------------------------
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7f68dada9ca5]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7f68dada9edc]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7f68dac83944]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7f68dad35c3e]
/lib/x86_64-linux-gnu/libc.so.6 [0x7f68da88b4b0]
/lib/x86_64-linux-gnu/libc.so.6(strlen+0x26) [0x7f68da8e0b96]
/lib/x86_64-linux-gnu/libc.so.6(getenv+0x16) [0x7f68da88f786]
[0x7f68cefc700b]
/usr/lib/x86_64-linux-gnu/libLLVM-3.6.so.1(_ZN4llvm5MCJIT11runFunctionEPNS_8FunctionERKSt6vectorINS_12GenericValueESaIS4_EE+0x132) [0x7f68d4d96c12]
/usr/lib/x86_64-linux-gnu/libLLVM-3.6.so.1(LLVMRunFunction+0x1fe) [0x7f68d55179ae]
/usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call_unix64+0x4c) [0x7f68d5fa7e40]
/usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call+0x2eb) [0x7f68d5fa78ab]
/var/lib/gems/2.3.0/gems/ffi-1.9.18/lib/ffi_c.so(rbffi_CallFunction+0xf0) [0x7f68d61b73b0] Call.c:430
/var/lib/gems/2.3.0/gems/ffi-1.9.18/lib/ffi_c.so(custom_trampoline+0x26) [0x7f68d61baaa6] MethodHandle.c:232
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7f68dad9750b]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7f68dad9b269]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7f68dada0142]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3(rb_yield+0x364) [0x7f68dada6cb4]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3(rb_ensure+0xb0) [0x7f68dac89370]
/var/lib/gems/2.3.0/gems/ffi-1.9.18/lib/ffi_c.so(memptr_initialize+0xc2) [0x7f68d61ba952] MemoryPointer.c:90
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7f68dada1fee]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7f68dada2b72]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3(rb_class_new_instance+0x21) [0x7f68dacd6461]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7f68dad9750b]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7f68dad9b19c]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7f68dada0142]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3(rb_yield+0x364) [0x7f68dada6cb4]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3(rb_ary_each+0x3d) [0x7f68dac44e4d]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7f68dad9750b]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7f68dad9b19c]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7f68dada0142]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3(rb_yield+0x364) [0x7f68dada6cb4]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7f68dac48a4d]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7f68dad9750b]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7f68dada54a3]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7f68dada64d3]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7f68dad9b19c]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7f68dada0142]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7f68dada0d35]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7f68dada0e13]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3(rb_proc_call+0x42) [0x7f68dac8fb12]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7f68dac86c24]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3(rb_exec_end_proc+0xbb) [0x7f68dac89f4b]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3 [0x7f68dac8a073]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3(ruby_cleanup+0x1fa) [0x7f68dac8a41a]
/usr/lib/x86_64-linux-gnu/libruby-2.3.so.2.3(ruby_run_node+0x25) [0x7f68dac8a845]
/usr/bin/ruby2.3 [0x40087b]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf0) [0x7f68da876830]
/usr/bin/ruby2.3(_start+0x29) [0x4008a9]